Tests/MessagingInApp/Gist/Managers/MessageQueueManagerTest.swift

```diff
@@ -64,6 +64,79 @@ class MessageQueueManagerTest: UnitTest {
         XCTAssertEqual(actualMessages, [givenMessage2, givenMessage1, givenMessage3])
     }
 
+    func test_getInlineMessages_givenQueueMessageHasPageRule_pageRuleMatch_expectInlineMessage() {
+        let givenElementId = String.random
+        let currentRoute = "^(Dashboard)$"
+
+        let givenMessage1 = Message(pageRule: currentRoute, elementId: givenElementId, priority: 0)
+        let givenMessage2 = Message(pageRule: currentRoute, elementId: givenElementId, priority: 1)
+        let givenMessage3 = Message(elementId: givenElementId, priority: nil)
+
+        manager.localMessageStore = [
+            "1": givenMessage1,
+            "2": givenMessage2,
+            "3": givenMessage3
+        ]
+        Gist.shared.setCurrentRoute("Dashboard")
+        let actualMessages = manager.getInlineMessages(forElementId: givenElementId)
+        XCTAssertEqual(actualMessages, [givenMessage1, givenMessage2, givenMessage3])
+    }
+
+    func test_getInlineMessages_givenQueueMessageHasPageRule_pageRuleMisMatch_expectInlineMessageWithoutPageRule() {
+        let givenElementId = String.random
+        let currentRoute = "^(Dashboard)$"
+
+        let givenMessage1 = Message(pageRule: currentRoute, elementId: givenElementId, priority: 0)
+        let givenMessage2 = Message(pageRule: currentRoute, elementId: givenElementId, priority: 1)
+        let givenMessage3 = Message(elementId: givenElementId, priority: nil)
+
+        manager.localMessageStore = [
+            "1": givenMessage1,
+            "2": givenMessage2,
+            "3": givenMessage3
+        ]
+        Gist.shared.setCurrentRoute("Home")
+        let actualMessages = manager.getInlineMessages(forElementId: givenElementId)
+        XCTAssertEqual(actualMessages, [givenMessage3])
+    }
+
+    func test_getInlineMessages_givenQueueMessageHasPageRule_pageRuleMisMatch_expectEmptyArray() {
+        let givenElementId = String.random
+        let currentRoute = "^(Dashboard)$"
+
+        let givenMessage1 = Message(pageRule: currentRoute, elementId: givenElementId, priority: 0)
+        let givenMessage2 = Message(pageRule: currentRoute, elementId: givenElementId, priority: 1)
+        let givenMessage3 = Message(pageRule: currentRoute, elementId: givenElementId, priority: nil)
+
+        manager.localMessageStore = [
+            "1": givenMessage1,
+            "2": givenMessage2,
+            "3": givenMessage3
+        ]
+        Gist.shared.setCurrentRoute("Home")
+        let actualMessages = manager.getInlineMessages(forElementId: givenElementId)
+        XCTAssertEqual(actualMessages, [])
+    }
+
+    func test_getInlineMessages_givenQueueMessageHasPageRule_givenNoSetCurrentRoute_expectEmptyArray() {
```

Comment:
I do think that it's important to test `givenNoSetCurrentRoute` as the function already does. I see the function is also testing the result when the elementIds are unique between all of the messages. 

To avoid testing multiple cases in 1 test function, I suggest having all of the messages use the same elementid and only test the result when no current route is set. 

I believe the function `test_getInlineMessages_givenQueueNotEmpty_expectGetInlineMessagesWithElementId` is testing the case when element ids do not match.

Reply:
Catered. ✅